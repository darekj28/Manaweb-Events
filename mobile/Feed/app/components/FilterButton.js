import React from 'react';
import {Component} from 'react'
import { AppRegistry,StyleSheet,Text,View,ListView,TouchableOpacity,TouchableHighlight, TextInput,
        TouchableWithoutFeedback, Alert, Image, Animated} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class FilterButton extends Component {
	constructor(props) {
		super(props);
		this.state = {isSelected : this.props.selected};
		this.handleClick = this.handleClick.bind(this);
	}
	handleClick() {
		this.setState({ isSelected : !this.state.isSelected });
		this.props.onClick(this.props.name, this.props.isSearch);
	}

	componentDidMount() {
		// $('[data-toggle="tooltip"]').tooltip(); 
	}

	render() {
		var icon;
		var selected = this.state.isSelected ? "icon-success" : "icon-danger";
		switch(this.props.name) {
			case 'Trade' : 
				icon="glyphicon glyphicon-transfer";
				break;
			case 'Play' :
				icon="glyphicon glyphicon-play";
				break;
			case 'Chill' :
				icon="glyphicon glyphicon-time";
				break;
			default : 
				al

				ert('Invalid action.');
		}
		if (!this.props.isSearch)
			return(

				<a className="input-group-addon" data-container="body" data-toggle="tooltip" title={this.props.name}>
					<span className={icon + " filterButton " + selected} onClick={this.handleClick}>
					</span>
				</a>
				)
		else return(<a className="input-group-addon" data-container="body" data-toggle="tooltip" title={this.props.name}
					 data-placement="bottom">
					<span className={icon + " filterButton " + selected} onClick={this.handleClick}>
					</span>
				</a>
			);
	}
}

FilterButton.defaultProps = {
	isSearch : false
};