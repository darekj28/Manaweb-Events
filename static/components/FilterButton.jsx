var React = require('react');
export default class FilterButton extends React.Component {
	constructor(props) {
		super(props);
		this.state = {isSelected : this.props.selected};
		this.handleClick = this.handleClick.bind(this);
	}
	handleClick() {
		var that = this;
		this.setState({ isSelected : !this.state.isSelected });
		this.props.onClick(that.props.name, that.props.isSearch);
	}

	componentDidMount() {
		$(document).ready(function(){
		    $('[data-toggle="tooltip"]').tooltip(); 
		});
	}

	render() {
		var icon;
		var selected = this.state.isSelected ? "icon-success" : "icon-danger";
		switch(this.props.name) {
			case 'Trade' : 
				icon="glyphicon glyphicon-transfer";
				break;
			case 'Play' :
				icon="glyphicon glyphicon-play";
				break;
			case 'Chill' :
				icon="glyphicon glyphicon-time";
				break;
			default : 
				alert('Invalid action.');
		}
		return(<a className="input-group-addon" data-toggle="tooltip" title= {this.props.name}>
			<span className={icon + " filterButton " + selected}
					onClick={this.handleClick}>
					</span>
			</a>);
	}
}

FilterButton.defaultProps = {
	isSearch : false
};