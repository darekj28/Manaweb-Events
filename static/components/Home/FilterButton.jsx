var React = require('react');
export default class FilterButton extends React.Component {
	constructor(props) {
		super(props);
		this.state = {isSelected : this.props.selected};
		this.handleClick = this.handleClick.bind(this);
	}
	handleClick() {
		this.setState({ isSelected : !this.state.isSelected });
		this.props.onClick(this.props.name, this.props.isSearch);
	}

	componentDidMount() {
		$('[data-toggle="tooltip"]').tooltip(); 
	}

	render() {
		var icon;
		var selected = this.state.isSelected ? "icon-success" : "icon-danger";
		switch(this.props.name) {
			case 'Trade' : 
				icon="glyphicon glyphicon-transfer";
				break;
			case 'Play' :
				icon="glyphicon glyphicon-play";
				break;
			case 'Chill' :
				icon="glyphicon glyphicon-time";
				break;
			default : 
				alert('Invalid action.');
		}
		if (!this.props.isSearch)
			return(<a className="input-group-addon" data-container="body" data-toggle="tooltip" title= {this.props.name}>
					<span className={icon + " filterButton " + selected} onClick={this.handleClick}>
					</span>
				</a>)
		else return(<a className="input-group-addon">
					<span className={icon + " filterButton " + selected} onClick={this.handleClick}>
					</span>
				</a>
			);
	}
}

FilterButton.defaultProps = {
	isSearch : false
};