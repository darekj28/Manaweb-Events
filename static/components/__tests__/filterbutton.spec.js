import FilterButton from "../FilterButton.jsx";

describe("FilterButton", () => {
	it("renders correctly for active", ()=> {
		const fb = renderer.create(
			<FilterButton name="Trade" onClick={jest.fn()} active={true} 
					isSearch={false}/> 
			);
		expect(fb).toMatchSnapshot();
	});
	it("renders correctly for inactive", ()=> {
		const fb = renderer.create(
			<FilterButton name="Trade" onClick={jest.fn()} active={false} 
					isSearch={false}/> 
			);
		expect(fb).toMatchSnapshot();
	});
	it("renders correctly for all actions", ()=> {
		const fb = renderer.create(
			<div>
				<FilterButton name="Trade" onClick={jest.fn()} active={false} 
						isSearch={false}/> 
				<FilterButton name="Play" onClick={jest.fn()} active={false} 
				isSearch={false}/> 
				<FilterButton name="Chill" onClick={jest.fn()} active={false} 
				isSearch={false}/> 
			</div>
			);
		expect(fb).toMatchSnapshot();
	});
	it("should call onClick with the name and isSearch when user clicks it", () => {
		expect(true).toBe(true);
	});
});