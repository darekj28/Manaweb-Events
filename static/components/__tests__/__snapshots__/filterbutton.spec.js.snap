exports[`FilterButton renders correctly for active 1`] = `
<a
  className="input-group-addon">
  <span
    className="glyphicon glyphicon-transfer icon-success filterButton"
    onClick={[Function]} />
</a>
`;

exports[`FilterButton renders correctly for all actions 1`] = `
<div>
  <a
    className="input-group-addon">
    <span
      className="glyphicon glyphicon-transfer icon-danger filterButton"
      onClick={[Function]} />
  </a>
  <a
    className="input-group-addon">
    <span
      className="glyphicon glyphicon-play icon-danger filterButton"
      onClick={[Function]} />
  </a>
  <a
    className="input-group-addon">
    <span
      className="glyphicon glyphicon-time icon-danger filterButton"
      onClick={[Function]} />
  </a>
</div>
`;

exports[`FilterButton renders correctly for inactive 1`] = `
<a
  className="input-group-addon">
  <span
    className="glyphicon glyphicon-transfer icon-danger filterButton"
    onClick={[Function]} />
</a>
`;
